internal/silk/decoder.go

```diff
@@ -1389,23 +1389,21 @@ func (d *Decoder) ltpSynthesis(
 	//                                 out[i] - \  out[i-k-1] * --------, 1.0)
 	//                                          /_               4096.0
 	//                                          k=0
-	var (
-		resPtr *float32
-		outVal float32
-	)
+	var outVal float32
 	for i := (j - pitchLags[s] - 2); i < out_end; i++ {
 		index := i + j
 
+		var resVal float32
+		var resIndex int
 		if index >= 0 {
 			if index >= len(res) {
 				continue
 			}
-
-			resPtr = &res[index]
-			*resPtr = out[index]
+			resVal = out[index]
+			resIndex = index
 		} else {
-			resPtr = &resLag[len(resLag)+index]
-			*resPtr = d.finalOutValues[len(d.finalOutValues)+index]
+			resIndex = len(resLag) + index
```

Comment:
I realized I made a mistake here, I'll fix a bit later.

Reply:
@egonelbre Is this fixed/ok to merge?

Reply:
Yes, it's fixed now.